crates/common/src/compile.rs

```diff
@@ -59,6 +61,9 @@ pub struct ProjectCompiler {
 
     /// Extra files to include, that are not necessarily in the project's source dir.
     files: Vec<PathBuf>,
+
+    /// Whether to compile with preprocessed cache for tests and scripts.
+    cache_tests: bool,
```

Comment:
imo this isn't 100% clear what cache_tests means because test artifacts are already cached, but this is different, so maybe we can come up with a name that is more appropriate, even if slightly more verbose?

Reply:
wdyt about one of `optimize_tests` / `optimize_tests_compilation` / `optimize_tests_compiler`?